src/adaXT/decision_tree/decision_tree.pyx

```diff
@@ -155,24 +155,28 @@ class DecisionTree(BaseModel):
         return ht
 
     def _tree_based_weights(self, hash0: dict, hash1: dict, size_X0: int,
-                            size_X1: int, scale: bool) -> np.ndarray:
+                            size_X1: int, scaling: str) -> np.ndarray:
         matrix = np.zeros((size_X0, size_X1))
```

Comment:
Given there is now an inner loop, it might a good idea to give the loop a type definition for performance.

Reply:
Yes, good point. Can you maybe propose something here. Also do you see an alternative of getting around the inner loop, I didn't manage to find anything so I stuck with this?